### PR TITLE
Add state message for mechanum controller

### PR DESCRIPTION
**Pull request by destogl**
Requirement for ros-controls/ros2_controllers#512

### CODE REVIEW

File: control_msgs/msg/MecanumDriveControllerState.msg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+std_msgs/Header header
+nav_msgs/Odometry odom
```
Comment (destogl) [inline control_msgs/msg/MecanumDriveControllerState.msg:2]:
```suggestion
nav_msgs/Odometry odometry
```

File: control_msgs/msg/MecanumDriveControllerState.msg [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+float64 front_left_wheel_velocity
+float64 back_left_wheel_velocity
+float64 back_right_wheel_velocity
+float64 front_right_wheel_velocity
+geometry_msgs/Twist reference_vel
```
Comment (destogl) [inline control_msgs/msg/MecanumDriveControllerState.msg:7]:
```suggestion
geometry_msgs/Twist reference_velocity
```

**Review by destogl — COMMENTED** [review]
This looks good now! Thanks!

**Comment by destogl** [comment]
@bmagyar what do you think that we name this controller **status* message instead of **state**.

File: control_msgs/CMakeLists.txt [hunk 3]
```diff
@@ -58,6 +60,7 @@ rosidl_generate_interfaces(${PROJECT_NAME}
     action_msgs
     builtin_interfaces
     geometry_msgs
+    nav_msgs
```
Comment (destogl) [inline control_msgs/CMakeLists.txt:63]:
```suggestion
```

File: control_msgs/msg/MecanumDriveControllerState.msg [hunk 4]
```diff
@@ -0,0 +1,7 @@
+std_msgs/Header header
+nav_msgs/Odometry odometry
```
Comment (destogl) [inline control_msgs/msg/MecanumDriveControllerState.msg:2]:
```suggestion
```

**Review by bmagyar — APPROVED** [review]

**Comment by bmagyar** [comment]
@mergifyio backport humble

**Pull request merged**